core/src/main/scala/spinal/core/internals/Phase.scala

```diff
@@ -286,158 +287,249 @@ class PhaseApplyIoDefault(pc: PhaseContext) extends PhaseNetlist{
   }
 }
 
-
-
 class PhaseAnalog extends PhaseNetlist{
 
   override def impl(pc: PhaseContext): Unit = {
     import pc._
 
-
-
-    val wraps = mutable.LinkedHashMap[BaseType, BaseType]()
-
-    //Identify every InOut(Analog) and create a wrapper in the parent component
-    walkComponents(c => if(c.parent != null) c.ioSet.withFilter(_.isInOut).foreach(io => {
-      val wrap = c.parent.rework{
-        Analog(io)
+    case class Bit(bt : BaseType, bitId : Int, scope : ScopeStatement)
+    // A island represent the connection of multiple analog single bits
+    class Island {
+      def absorbe(other: Island): Unit = {
+        this.elements ++= other.elements
       }
-      wraps += io -> wrap
-    }))
-
-    //Remap all driving expression to the created wrapper
-    GraphUtils.walkAllComponents(topLevel, c => c.dslBody.walkStatements(s => s.walkRemapDrivingExpressions{e => e match {
-      case source : BaseType => wraps.get(source) match {
-        case Some(target) if c == target.component => target
-        case _ => e
+      val elements = mutable.LinkedHashSet[Bit]()
+      def add(that : Bit): Unit ={
+        elements += that
       }
-      case _ => e
-    }}))
-
-    //Connect the wrappers
-    for((source, target) <- wraps){
-      target.component.rework(target := source)
-    }
-
-    //Be sure that sub io assign parent component stuff
-    walkComponents(c => c.ioSet.withFilter(_.isInOut).foreach(io => {
-      io.foreachStatements {
-        case s@AssignmentStatement(_: BaseType, x: BaseType) if x.isAnalog && x.component == c.parent =>
-          s.dlcRemove()
-          x.dlcAppend(s)
-          s.target = x
-          s.source = io
-        case _ =>
-      }
-    }))
-
-    val analogs        = ArrayBuffer[BaseType]()
-    val islands        = mutable.LinkedHashSet[mutable.LinkedHashSet[BaseType]]()
-    val analogToIsland = mutable.HashMap[BaseType,mutable.LinkedHashSet[BaseType]]()
-
-    def addToIsland(that: BaseType, island: mutable.LinkedHashSet[BaseType]): Unit = {
-      island += that
-      analogToIsland(that) = island
     }
 
-    //val wrapped = mutable.HashMap[BaseType, BaseType]()
 
-    walkStatements{
-      case bt: BaseType if bt.isAnalog =>
-        analogs += bt
-
-        //Manage islands
-        bt.foreachStatements {
-          case s@AssignmentStatement(x, y: BaseType) if y.isAnalog =>
-            if (s.finalTarget.component == y.component) {
-              (analogToIsland.get(bt), analogToIsland.get(y)) match {
+    pc.walkComponents { c =>
+      val islands = mutable.LinkedHashSet[Island]()
+      val bitToIsland = mutable.HashMap[Bit, Island]()
+
+      // Generate the list of islands
+      c.dslBody.walkStatements {
+        case s: AssignmentStatement => {
+          val targetBt = s.finalTarget
+          val sourceBt : BaseType = s.source match {
+            case bt: BaseType => bt
```

Comment:
So, there is first something else.

myAnalogBits(0) will return a non analog Bool()
So 
myAnalogBool := myAnalogBits(0) 

Will drive myAnalogBool with the readed value of myAnalogBits(0) , It will not do a analog connection.

myAnalogBool := myAnalogBits(0).setAsAnalog() will do a analog connection

So yeah, to resume the current state : 
```scala
myAnalogBits(range) := ??? will do a analog connection
??? := myAnalogBits(range) will read the analog value of myAnalogBits and use it to drive ???
??? := myAnalogBits(range).setAsAnalog()) will do a analog connection
```

So there is some logic behind it.
Not sure actualy if that's what we want.

Then when this is established, BitsBitAccessFloating should not be allowed on analog i think

Reply:
I pushed an update to have : 

```scala
myAnalogBits(range) := ??? will do a analog connection
??? := myAnalogBits(range) will do a analog connection
??? := myAnalogBits(range).setAsComb() will read myAnalogBits(range). to drive ???
```

I think that's less surprising like this.